zookeeper-api/src/test/java/org/apache/helix/zookeeper/impl/client/TestSharedZkClient.java

```diff
@@ -40,7 +40,7 @@ public void beforeClass() throws IOException, InvalidRoutingDataException {
     _realmAwareZkClientFactory = SharedZkClientFactory.getInstance();
   }
 
-  @Test
+  @Test(dependsOnMethods = "testRealmAwareZkClientCreation")
```

Comment:
Is there any point on keeping this test case at a different class? If it depends on _realmAwareZkClient that has to be created, then maybe this should just go to RealmAwareZkClientFactoryTestBase since that's what this is testing.

Reply:
`RealmAwareZkClientFactoryTestBase` is extended in both `TestSharedZkClient` and `TestDedicatedZkClient`. Putting this method in the base class will cause duplicated test and logic change in its other subclass. So I think it's better to keep it this way. Thanks.